kloppy/infra/serializers/tracking/skillcorner.py

```diff
@@ -0,0 +1,427 @@
+                player["trackable_object"]: player
+                for player in metadata["players"]
+            }
+
+            referee_dict = {
+                ref["trackable_object"]: "referee"
+                for ref in metadata["referees"]
+            }
+            ball_id = metadata["ball"]["trackable_object"]
+
+            # there are different pitch_sizes in SkillCorner
+            pitch_size_width = metadata["pitch_width"]
+            pitch_size_length = metadata["pitch_length"]
+
+            home_team_id = metadata["home_team"]["id"]
+            away_team_id = metadata["away_team"]["id"]
+
+            players = {"HOME": {}, "AWAY": {}}
+
+            home_team = Team(
+                team_id=home_team_id,
+                name=metadata["home_team"]["name"],
+                ground=Ground.HOME,
+            )
+            self.home_team = home_team
+
+            away_team = Team(
+                team_id=away_team_id,
+                name=metadata["away_team"]["name"],
+                ground=Ground.AWAY,
+            )
+            self.away_team = away_team
+
+            teams = [home_team, away_team]
+
+            for player_id in player_dict.keys():
+                player = player_dict.get(player_id)
+                team_id = player["team_id"]
+
+                if team_id == home_team_id:
+                    team_string = "HOME"
+                    team = home_team
+                elif team_id == away_team_id:
+                    team_string = "AWAY"
+                    team = away_team
+
+                players[team_string][player_id] = Player(
+                    player_id=f"{team.ground}_{player['number']}",
+                    team=team,
+                    jersey_no=player["number"],
+                    name=" ".join([player["first_name"], player["last_name"]]),
+                    first_name=player["first_name"],
+                    last_name=player["last_name"],
+                    starting=player["start_time"] == "00:00:00",
+                    position=Position(
+                        position_id=player["player_role"].get("id"),
+                        name=player["player_role"].get("name"),
+                        coordinates=None,
+                    ),
+                    attributes={},
+                )
+
+            home_team.players = list(players["HOME"].values())
+            away_team.players = list(players["AWAY"].values())
+
+        anon_players = {"HOME": {}, "AWAY": {}}
+
+        with performance_logging("Loading data", logger=logger):
+
+            def _iter():
+                n = 0
+                sample = 1.0 / sample_rate
+
+                for frame in raw_data:
+                    frame_period = frame["period"]
+
+                    if frame_period is not None:
+                        if n % sample == 0:
+                            yield frame
+                        n += 1
+
+        frames = []
+        for n, _frame in enumerate(_iter()):
```

Comment:
Did you consider moving the logic from within `_iter` to within this loop?

Reply:
What's the advantage of doing so?

I oriented this after the TRACABSerializer, which does it similarly.

Reply:
Ah I missed the sample rate code here. Yes. Let's keep it your way.